packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/albums/CreateAlbum.kt

```diff
@@ -0,0 +1,69 @@
+package expo.modules.medialibrary.albums
+
+import android.content.Context
+import android.os.AsyncTask
+import android.media.MediaScannerConnection
+import android.net.Uri
+import android.provider.MediaStore
+import expo.modules.core.Promise
+import expo.modules.medialibrary.MediaLibraryConstants.*
+import expo.modules.medialibrary.MediaLibraryUtils
+import expo.modules.medialibrary.ifNull
+import java.io.File
+import java.io.IOException
+
+internal class CreateAlbum(
+  private val context: Context,
+  private val albumName: String,
+  private val assetId: String,
+  copyAsset: Boolean,
+  private val promise: Promise
+) : AsyncTask<Void?, Void?, Void?>() {
+  private val mStrategy = if (copyAsset) AssetFileStrategy.copyStrategy else AssetFileStrategy.moveStrategy
+
+  private fun createAlbum(mimeType: String): File? {
+    val albumDir = MediaLibraryUtils.getEnvDirectoryForAssetType(mimeType, false)
+      .ifNull {
+        promise.reject(ERROR_NO_ALBUM, "Could not guess asset type.")
+        return null
+      }
+
+    val album = File(albumDir.path, albumName)
+      .takeIf { it.exists() || it.mkdirs() }
+      .ifNull {
+        promise.reject(ERROR_NO_ALBUM, "Could not create album directory.")
+        return null
+      }
+    return album
+  }
+
+  public override fun doInBackground(vararg params: Void?): Void? {
```

Comment:
The type of this function looks stupid 😛  But we can't do anything with it right now.